src/backend/distributed/commands/multi_copy.c

```diff
@@ -2743,6 +2755,60 @@ FinishLocalCopy(CitusCopyDestReceiver *copyDest)
 }
 
 
+/*
+ * StartLocalColocatedIntermediateFile creates a co-located file for the given
+ * shard, and appends the binary headers if needed.
+ */
+static void
+StartLocalColocatedIntermediateFile(CitusCopyDestReceiver *copyDest,
+									CopyShardState *shardState)
+{
+	/* make sure the directory exists */
+	CreateIntermediateResultsDirectory();
+
+	const int fileFlags = (O_APPEND | O_CREAT | O_RDWR | O_TRUNC | PG_BINARY);
```

Comment:
what is `PG_BINARY `? In general, double check these flags, it is copy-paste

Reply:
We probably shouldn't have O_APPEND here, O_TRUNC will already truncate, is there any benefit?

Reply:
It looks like `PG_BINARY == 0` on non-windows. Given that we do not support windows, I'm removing it for simplicity:  https://github.com/postgres/postgres/blob/ca3b37487be333a1d241dab1bbdd17a211a88f43/src/include/c.h#L1246-L1263

Reply:
> We probably shouldn't have O_APPEND here, O_TRUNC will already truncate, is there any benefit?

And, also we create the file here, so `O_APPEND` is not correct anyway.

Reply:
Thanks